drivers/serial/uart_native_posix.c

```diff
@@ -196,12 +305,35 @@ static int open_tty(struct native_uart_status *driver_data,
 		close(open(slave_pty_name, O_RDWR | O_NOCTTY));
 	}
 	if (do_auto_attach) {
-		attach_to_tty(slave_pty_name);
+		if (auto_attach_cmd == NULL) {
+			auto_attach_cmd = (char *)default_cmd;
+		}
+
+		/* Roll our own %s substitution as sprintf() is too risky */
```

Comment:
Why risky? You can use snprintf() if concerned about overflowing the buffer.

Reply:
Hi @dcpleung thanks for the review. I'm concerned about the security of using sprintf() or snprintf() with an arbitrary and possibly malicious format string.

Consider the following code (you'll probably get a warning if you compile it but it should work on most architectures, possibly with some tweaking for more exotic ones)
```
#include <stdio.h>
#include <stdlib.h>
main()
{
   int secret = rand();
   printf("%p %p %p %p %p %p\n");
   fprintf(stderr, "secret=0x%x\n", secret);
}
```
this dumps part of the stack, leaking to stdout a secret intended for stderr.

Since we want to substitute the pty name at most once into a command string, it seemed safer to continue to use the sprintf() format but only implement it for a single string type.

Reply:
Hi @dcpleung thanks for the review. I'm concerned about the security of using sprintf() or snprintf() with an arbitrary and possibly malicious format string.

Consider the following code (you'll probably get a warning if you compile it but it should work on most architectures, possibly with some tweaking for more exotic ones)
```
#include <stdio.h>
#include <stdlib.h>
main()
{
   int secret = rand();
   printf("%p %p %p %p %p %p\n");
   fprintf(stderr, "secret=0x%x\n", secret);
}
```
this dumps part of the stack, leaking to stdout a secret intended for stderr.

Since we want to substitute the pty name at most once into a command string, it seemed safer to continue to use the sprintf() format but only implement it for a single string type.

Reply:
If this is the case, please explain in the comment. It helps the next person reading this code.

Reply:
@Chris-StJohn-Bose Note that this is a test setup, build and run by the user with the user permissions. If the user wanted to do anything malicious, he could just do it directly ;).